featuretests/api_credentialmanager_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package featuretests
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/api/credentialmanager"
+	"github.com/juju/juju/juju/testing"
+)
+
+// This suite only exists because no user facing calls exercise
+// invalidate credential calls enough to expose serialisation bugs.
+// If/when we have commands that would expose this,
+// we should drop this suite and write a new command-based one.
+
+type CredentialManagerSuite struct {
+	testing.JujuConnSuite
+	client *credentialmanager.Client
+}
+
+func (s *CredentialManagerSuite) SetUpTest(c *gc.C) {
+	s.JujuConnSuite.SetUpTest(c)
+
+	info := s.APIInfo(c)
+	userConn := s.OpenAPIAs(c, info.Tag, info.Password)
+
+	s.client = credentialmanager.NewClient(userConn)
+}
+
+func (s *CredentialManagerSuite) TearDownTest(c *gc.C) {
+	s.client.Close()
+	s.JujuConnSuite.TearDownTest(c)
+}
+
+func (s *CredentialManagerSuite) TestInvalidateModelCredential(c *gc.C) {
+	err := s.client.InvalidateModelCredential("no reason really")
```

Comment:
Since this is a feature test (end-end), the test should check that the state was actually updated as expected

Reply:
That's a great idea \o/ added :D Thank you for suggestion!